libnavigation-base/src/main/java/com/mapbox/navigation/base/route/NavigationRoute.kt

```diff
@@ -5,28 +5,192 @@ package com.mapbox.navigation.base.route
  * @param routeIndex the index of the route that this wrapper tracks
  * from the collection of routes returned in the original response.
  * @param routeOptions options used to generate the [directionsResponse]
- * @param id unique ID of this route entity used for matching it against additional metadata.
- * Unique ID doesn't mean that the underlying [directionsRoute] is unique.
- * Multiple [NavigationRoute]s can represent similar or equal [DirectionsRoute].
  */
-class NavigationRoute(
+class NavigationRoute internal constructor(
     val directionsResponse: DirectionsResponse,
     val routeIndex: Int,
     val routeOptions: RouteOptions,
-    val id: String = UUID.randomUUID().toString()
+    internal val nativeRoute: RouteInterface,
 ) {
 
+    companion object {
+        private val jobControl = InternalJobControlFactory.createDefaultScopeJobControl()
+
+        /**
+         * Creates new instances of [NavigationRoute] based on the routes found in the [directionsResponse].
+         *
+         * @param directionsResponse response to be parsed into [NavigationRoute]s
+         * @param routeOptions options used to generate the [directionsResponse]
+         * @param callback to get notified with the result of parsing
+         */
+        @JvmStatic
+        fun create(
+            directionsResponse: DirectionsResponse,
+            routeOptions: RouteOptions,
+            callback: OnNavigationRoutesReadyCallback
+        ): Job {
+            return jobControl.scope.launch {
+                val navigationRoutes = create(
+                    directionsResponse,
+                    directionsResponseJson = directionsResponse.toJson(),
+                    routeOptions,
+                    routeOptionsUrlString = routeOptions.toUrl("").toString()
+                )
+                callback.onNavigationRoutesReady(navigationRoutes)
+            }
+        }
+
+        /**
+         * Creates new instances of [NavigationRoute] based on the routes found in the [directionsResponseJson].
+         *
+         * @param directionsResponseJson response to be parsed into [NavigationRoute]s
+         * @param routeRequestUrl URL used to generate the [directionsResponse]
+         * @param callback to get notified with the result of parsing
+         */
+        @JvmStatic
+        fun create(
+            directionsResponseJson: String,
+            routeRequestUrl: String,
+            callback: OnNavigationRoutesReadyCallback
+        ): Job {
+            return jobControl.scope.launch {
+                val navigationRoutes = create(
+                    DirectionsResponse.fromJson(directionsResponseJson),
+                    directionsResponseJson = directionsResponseJson,
+                    RouteOptions.fromUrl(URL(routeRequestUrl)),
+                    routeOptionsUrlString = routeRequestUrl,
+                )
+                callback.onNavigationRoutesReady(navigationRoutes)
+            }
+        }
```

Comment:
why not just a suspend function?
```suggestion
        @JvmStatic
        suspend fun create(
            directionsResponseJson: String,
            routeRequestUrl: String
        ): NavigationRoute {
            return withContext(Dispatchers.Default) {
                create(
                    DirectionsResponse.fromJson(directionsResponseJson),
                    directionsResponseJson = directionsResponseJson,
                    RouteOptions.fromUrl(URL(routeRequestUrl)),
                    routeOptionsUrlString = routeRequestUrl,
                )
            }
        }
```